libgadget/init.c

```diff
@@ -265,50 +264,51 @@ setup_smoothinglengths(int RestartSnapNum, DomainDecomp * ddecomp)
 
         u_init /= molecular_weight;
 
-#ifdef DENSITY_INDEPENDENT_SPH
-        /* initialization of the entropy variable is a little trickier in this version of SPH,
-           since we need to make sure it 'talks to' the density appropriately */
-        message(0, "Converting u -> entropy, with density split sph\n");
+        if(All.DensityIndependentSphOn) {
```

Comment:
Does it make sense to move this as another function? The processing is quite long for a if branch.

Reply:
Done!